pkg/provisioner/ironic/ironic.go

```diff
@@ -1613,6 +1617,30 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 		return p.setMaintenanceFlag(ironicNode, true)
 	}
 
+	if ironicNode.PowerState == powerOn {
+		_, err := p.hardPowerOff()
+		if err != nil {
+			result.Dirty = true
+			switch err.(type) {
+			case HostLockedError:
+				p.log.Info("could not power off host, busy")
+				return operationContinuing(powerRequeueDelay)
+			default:
+				// For other errors continue with the delete
+				if p.host.Status.ErrorCount > maxErrorRetryCount {
+					result.Dirty = false
```

Comment:
not required

Reply:
Dirty flag is set to true as soon as we detect there is an error (here https://github.com/metal3-io/baremetal-operator/pull/412/files#diff-f05fc4a0c9679815bc34bcef100c2a8d525688f2804a1128c5b9e1e575b76c29R1623). When we don't want to try again, we need to reset it, right?

Reply:
At the end of the flow having the `dirty` set to true indicates to update the current host status and to continue. In this case the intention, IIUC, is to give up and stop with the reconciliation. `operationComplete` it's used for that (and in fact it does not set the dirty flag)

Reply:
Ok. Understand. I removed setting of the dirty flag here.